.gitlab-ci.yml

```diff
@@ -0,0 +1,93 @@
+# SPDX-License-Identifier: AGPL-3.0+
+#
+# Copyright (C) 2018 Ultimaker B.V.
+# Copyright (C) 2018 Olliver Schinagl <oliver@schinagl.nl>
+#
+# GitLab CI package build is split up into two parts.
+# * Build environment: creates the docker image that is needed for package generation
+# * Package generation: creates the final artifact
+
+stages:
+- lint
+- prepare
+- verify
+- push
+- build
+- cleanup
+
+# Common requisites
+# ==========
+.parameters_template: &parameters_common
+  tags:
+  - embedded
+  - docker
+
+lint build environment test:
```

Comment:
In general is suggest adding comments to each step defining its intent.

Reply:
done; note that the job name is also used in the UI, so I agree that this should be as descriptive as possible.